ios/browser/api/brave_shields/adblock_service.mm

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/brave_shields/adblock_service.h"
+
+#include "base/strings/sys_string_conversions.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/adblock_rust_ffi/src/wrapper.h"
+#include "brave/components/brave_component_updater/browser/dat_file_util.h"
+#include "brave/components/brave_shields/browser/ad_block_component_installer.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/ios/browser/api/brave_shields/adblock_filter_list+private.h"
+#include "components/component_updater/component_updater_service.h"
+
+@interface AdblockService () {
+  component_updater::ComponentUpdateService* _cus;  // NOT OWNED
+}
+@property(nonatomic, copy) NSString* shieldsInstallPath;
+@property(nonatomic, copy) NSArray<AdblockFilterList*>* regionalFilterLists;
+@end
+
+@implementation AdblockService
+
+- (instancetype)initWithComponentUpdater:
+    (component_updater::ComponentUpdateService*)componentUpdaterService {
+  if ((self = [super init])) {
+    _cus = componentUpdaterService;
+  }
+  return self;
+}
+
+- (void)registerDefaultShieldsComponent {
+  __weak auto weakSelf = self;
+  brave_shields::RegisterAdBlockDefaultComponent(
+      _cus, base::BindRepeating(^(const base::FilePath& install_path) {
+        [weakSelf adblockComponentDidBecomeReady:install_path];
+      }));
+}
+
+- (void)adblockComponentDidBecomeReady:(const base::FilePath&)install_path {
+  auto component_path = install_path;
+
+  // Update shields install path (w/ KVO)
+  [self willChangeValueForKey:@"shieldsInstallPath"];
+  self.shieldsInstallPath = base::SysUTF8ToNSString(install_path.value());
+  [self didChangeValueForKey:@"shieldsInstallPath"];
+
+  // Get filter lists from catalog
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock()},
+      base::BindOnce(&brave_component_updater::GetDATFileAsString,
+                     component_path.AppendASCII("regional_catalog.json")),
+      base::BindOnce(^(const std::string json) {
+        auto catalog = brave_shields::RegionalCatalogFromJSON(json);
+
+        NSMutableArray* lists = [[NSMutableArray alloc] init];
+        for (const auto& list : catalog) {
+          [lists addObject:[[AdblockFilterList alloc]
+                               initWithFilterList:adblock::FilterList(list)]];
+        }
+        self.regionalFilterLists = lists;
+
+        if (self.shieldsComponentReady) {
```

Comment:
Do we need `weakSelf`?

Reply:
Technically we don't, since we're not actually holding onto the closure technically it will just keep `AdblockService` alive until the callback is finished. I'll add it anyways however since it doesn't matter to keep it alive

Reply:
Done in https://github.com/brave/brave-core/pull/14248/commits/3d88a54b96aa61395da05e7196b98d3fc3337165